circuit-lib/circuit-lib.circom/benchmarks/index.ts

```diff
@@ -0,0 +1,319 @@
+        0,
+        1,
+        3,
+        2,
+        2
+    );
+        `;
+    const circuitName = `transaction${nInputUtxos}in${nOutputUtxos}out${
+      this.app ? "App" : "Masp"
+    }MTH${merkleTreeHeight}`;
+    this.circuitModels.push({
+      name: circuitName,
+      merkleTreeHeight,
+      nInputUtxos,
+      nOutputUtxos,
+    });
+    const finalPath =
+      this.circuitParameterVariants.outputPath + "/" + circuitName + ".circom";
+
+    // Check if file already exists
+    if (!this.rewrite && existsSync(finalPath)) {
+      return;
+    }
+
+    // Write the generated content to the outputPath
+    writeFileSync(finalPath, circomTemplate.trim());
+  }
+
+  async batchGenerateCircomFileVariants() {
+    const {
+      merkleTreeHeightVariants,
+      nInputUtxosVariants,
+      nOutputUtxosVariants,
+    } = this.circuitParameterVariants;
+
+    await this.ensureDirectoryExists(this.circuitParameterVariants.outputPath);
+
+    const parameterVariants = this.combineArrays(
+      merkleTreeHeightVariants,
+      nInputUtxosVariants,
+      nOutputUtxosVariants,
+    );
+
+    for (let variant of parameterVariants) {
+      this.generateCircomFile(variant[0], variant[1], variant[2]);
+    }
+  }
+
+  private addPrivateKey(tx: Transaction, account: Account, seed32: string) {
+    const account_privateKey = Account.generateShieldedPrivateKey(
+      seed32,
+      tx.params.poseidon,
+    );
+    const account_publicKey = Account.generateShieldedPublicKey(
+      account_privateKey,
+      tx.params.poseidon,
+    );
+
+    tx.proofInput["inPrivateKey"] = tx.params.inputUtxos.map((utxo: Utxo) => {
+      if (
+        utxo.publicKey.eq(account.pubkey) &&
+        account.pubkey.eq(account_publicKey)
+      ) {
+        return account_privateKey;
+      }
+      if (STANDARD_SHIELDED_PUBLIC_KEY.eq(utxo.publicKey)) {
+        return STANDARD_SHIELDED_PRIVATE_KEY;
+      }
+    });
+  }
+
+  // for now the benchmarks are solely dependant on the Merkle Tree Height
+  async generateProofInputs(merkleTreeHeight: number) {
+    const lightProvider = await LightProvider.loadMock();
+    const poseidon = await buildPoseidonOpt();
+
+    const seed32 = bs58.encode(new Uint8Array(32).fill(1));
+    const account = new Account({ poseidon: poseidon, seed: seed32 });
+    await account.getEddsaPublicKey();
+
+    const depositAmount = 20_000;
+    const depositFeeAmount = 10_000;
+
+    const deposit_utxo1 = new Utxo({
+      poseidon: poseidon,
+      assets: [FEE_ASSET, MINT],
+      amounts: [new BN(depositFeeAmount), new BN(depositAmount)],
+      publicKey: account.pubkey,
+      assetLookupTable: lightProvider.lookUpTables.assetLookupTable,
+      verifierProgramLookupTable:
+        lightProvider.lookUpTables.verifierProgramLookupTable,
+    });
+
+    const mockPubkey: PublicKey = SolanaKeypair.generate().publicKey;
+
+    let txParams = new TransactionParameters({
+      outputUtxos: [deposit_utxo1],
```

Comment:
You didn't add in utxos yet, so you are still hashing zeros but you probably know that.

Reply:
yes, that's why I didn't resolve it from the earlier requested changes.

Reply:
done!